src/crimson/admin/admin_socket.cc

```diff
@@ -0,0 +1,896 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+/*
+ * Ceph - scalable distributed file system
+ *
+ * Copyright (C) 2011 New Dream Network
+ *
+ * This is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License version 2.1, as published by the Free Software
+ * Foundation.  See file COPYING.
+ *
+ */
+
+#include "common/version.h"
+#include "crimson/net/Socket.h"
+// activate for some inline unit-testing: #define UNIT_TEST_OSD_ASOK
+#include "crimson/admin/admin_socket.h"
+#include "crimson/common/log.h"
+//#include "seastar/testing/test_case.hh"
+#include "seastar/net/api.hh"
+#include "seastar/net/inet_address.hh"
+#include "seastar/core/reactor.hh"
+#include "seastar/core/thread.hh"
+//#include "seastar/util/log.hh"
+#include "seastar/util/std-compat.hh"
+#include <boost/algorithm/string.hpp>
+
+/*!
+  A Crimson-wise version of the admin socket - implementation file
+
+  \todo handle the unlinking of the admin socket. Note that 'cleanup_files' at-exit functionality is not yet
```

Comment:
CodingStyle > https://google.github.io/styleguide/cppguide.html#Line_Length

'Each line of text in your code should be at most 80 characters long.'

Reply:
will fix